curation-migration-datasync/eventConsumer.ts

```diff
@@ -55,3 +57,38 @@ export async function removeScheduledItem(
   // Remove association from DynamoDB
   await curatedItemModel.deleteByCuratedRecId(curatedRecord.curatedRecId);
 }
+
+/**
+ * Update the scheduled item in the legacy database based on the curatedRecId
+ * from the DynamoDB. If the DynamoDB mapping is not found, treat the event
+ * the same as an add-scheduled-item event.
+ * @param eventBody
+ * @param db
+ */
+export async function updateScheduledItem(
+  eventBody: ScheduledItemPayload,
+  db: Knex
+) {
+  const curatedItemModel = new CuratedItemRecordModel();
+  const scheduledItem = await curatedItemModel.getByScheduledItemExternalId(
+    eventBody.scheduledItemExternalId
+  );
+
+  if (!scheduledItem) {
+    const errorMessage = `update-scheduled-item: No mapping found for scheduledItemExternalId=${eventBody.scheduledItemExternalId}`;
+    Sentry.captureMessage(errorMessage);
+    console.log(errorMessage);
+    await addScheduledItem(eventBody, db);
```

Comment:
nice! coz the `scheduledItemEventPayload` gets all the required fields, https://github.com/Pocket/curated-corpus-api/blob/main/src/events/eventBus/EventBusConfig.ts#L58-L59

incase if the scheduledItem is missing in dynamo, we can just treat it as a new item and add them.